### ISSUE TITLE
Using this PDF viewer with Mendeley

### ISSUE THREAD

**Issue by A3EKV** [ahrm/sioyek#42]
Hello,
I would like to thank you for this piece of software: very helpful!
However, AFAIK, Most researchers are using some tools such as Mendeley to manage their documents/citations, etc.

Personally, I find it hard to transition from Mendeley to Sioyek and give up all the notes/highlight of each paper in my database of papers.

So, if Sioyek offers a solution to this problem, a lot of Mendeley will happily transition but It is still not clear how given that sioyek is not a library management software.

Isn't that possible? 
Thank you

**Comment by ahrm** [comment]
As far as I am aware Mendeley doesn't have a documented way to export annotations so I don't think that is possible.

**Comment by Endle** [comment]
I have a (a bit OT) suggest: As @ahrm mentioned in https://github.com/ahrm/sioyek/issues/29#issuecomment-915982417

> We also added the ability to import/export your data into a json file by entering the :import and :export commands respectively.

How about adding documents for importing and exporting?

Thanks

**Comment by s1291** [comment]
@ahrm 
Actually, Mendeley provides a REST API. You can easily import all your annotations. (see: https://api.mendeley.com/apidocs/docs). If I am not mistaken they have a Python and Javascript SDKs.
I can provide some examples if you are interested.

**Comment by ahrm** [comment]
> I have a (a bit OT) suggest: As @ahrm mentioned in [#29 (comment)](https://github.com/ahrm/sioyek/issues/29#issuecomment-915982417)
> 
> > We also added the ability to import/export your data into a json file by entering the :import and :export commands respectively.
> 
> How about adding documents for importing and exporting?
> 
> Thanks

Yes I should add documentation for import/export function but I don't think it will help with Mendeley because I think it uses a completely different file format.

>Actually, Mendeley provides a REST API. You can easily import all your annotations. (see: https://api.mendeley.com/apidocs/docs). If I am not mistaken they have a Python and Javascript SDKs.
I can provide some examples if you are interested.

Thanks I'll look into it.

**Comment by ahrm** [comment]
> @ahrm
> Actually, Mendeley provides a REST API. You can easily import all your annotations. (see: https://api.mendeley.com/apidocs/docs). If I am not mistaken they have a Python and Javascript SDKs.
> I can provide some examples if you are interested.

This looks good. However, I don't I will include it as part of sioyek itself. But I think I can write a script to convert this to some file format that can be imported by sioyek.

**Comment by s1291** [comment]
@ahrm 
The default format is JSON.
Another interesting idea is the ability to integrate the highlights back (from Sioyek to mendeley database) via POST request.

**Comment by ahrm** [comment]
> Doesn't Mendeley actually exports all pdfs with annotations that you made in their reader? You would get the highlights but would lose the annotations I imagine.

Yes but those highlights are embedded in the PDF and are different from sioyek highlights (for example you can not search them).

**Comment by qsh-zh** [comment]
By the way, does anyone know how to set the default pdf reviewer software for Mendeley when we open pdf externally?

**Comment by koppor** [comment]
@A3EKV If you like open source, you could consider switching to [JabRef](https://github.com/jabref/jabref/). It also provides management of references - and allows for configuring the external PDF reader.
